system/src/system_network_diagnostics.cpp

```diff
@@ -298,7 +303,8 @@ class NetworkCellularCellGlobalIdentityMobileCountryCodeDiagnosticData
 
     virtual int get(IntType& val)
     {
-        const CellularGlobalIdentity* cgi = s_networkCache.getCGI();
+        const CellularGlobalIdentity* const cgi = s_networkCache.getCGI();
+        CHECK_TRUE(cgi, SYSTEM_ERROR_NETWORK);
```

Comment:
Instead of returning CellularGlobalIdentity, how about passing it into getCGI by reference and returning the system error code directly.  There are a lot of specific errors that could be bubbled up instead of SYSTEM_ERROR_NETWORK, and this is kind of the same way cellular_global_identity() works already.